assets/js/atomic/blocks/product/button/block.js

```diff
@@ -101,7 +101,9 @@ const AddToCartButton = ( { product } ) => {
 		buttonProps.href = permalink;
 		buttonProps.rel = 'nofollow';
 	} else {
-		buttonProps.onClick = addToCart;
+		buttonProps.onClick = () => {
+			addToCart();
+		};
```

Comment:
Took me some time to understand why this was an issue when assigning `addToCart` directly to `onClick` we're also passing the args from `onClick` to `addToCart`, `addToCart` expects a quantity `int` as the first variables (defaults to 10), `onClick` passes an `event` object, causing this issue.
By calling `addToCart` on the callback, we avoid this.